### PR TITLE
Update syntax to respect TF minimum version (1.0.0)

### LINKED ISSUE #11652: Update terraform examples to match new syntax
## Details
The syntax used by the Teleport Terraform provider has changed for v9 and above, and needs updating

Per @marcoandredinis: 

> We also need to update the .tf example file because it's using the old syntax
We need to add = after metadata and spec keywords
There's also some error in the expires section, not sure yet what it is but I was only able to run the terraform apply after removing it

### Category

- Improve Existing
- Remove Outdated

### PR DESCRIPTION
**Pull request by marcoandredinis**
Most of the changes are related to the strictness of TF v0.12

https://www.terraform.io/language/upgrade-guides/0-12
> Due to the design of the configuration language decoder in
> Terraform v0.11 and earlier, it was in many cases possible to
> interchange the argument syntax (with =) and the block syntax
> (with just braces) when dealing with map arguments vs. nested
> blocks. However, this led to some subtle bugs and limitations,
> so Terraform v0.12 now requires consistent usage of argument
> syntax for arguments and nested block syntax for nested blocks.

This commit validates both examples against Terraform 1

The min terraform version is going to be updated to 1.0.0
https://github.com/gravitational/teleport/pull/11650

Fixes #11652

### CODE REVIEW

**Review by r0mant — APPROVED** [review]
lgtm but please wait for approval from @tcsc as well

**Review by ravicious — APPROVED** [review]

**Review by tcsc — APPROVED** [review]

**Pull request merged**
